src/test/java/teammates/test/driver/TestProperties.java

```diff
@@ -137,6 +139,45 @@ public static String extractVersionNumber(String inputString) {
         return inputString.substring(startPos, endPos).replace("-", ".").trim();
     }
     
-    
+    /**
+     * Verifies that the test properties specified in test.properties file allows for HTML
+     * regeneration via God mode to work smoothly (i.e all test HTML files are correctly regenerated, 
+     * strings that need to be replaced with placeholders are correctly replaced, and 
+     * strings that are not supposed to be replaced with placeholders are not replaced).
+     */
+    public void verifyReadyForGodMode() {
+        if (!inst().isDevServer()) {
+            Assumption.fail("God mode regeneration works only in dev server.");
+        }
+        if (!areTestAccountsReadyForGodMode()) {
+            Assumption.fail("Please append a unique id to each of the default account in test.properties "
+                            + "in order to use God mode, e.g change alice.tmms to alice.tmms.yourName, "
+                            + "charlie.tmms to charlie.tmms.yourName, etc.");
+        }
+        if (!areAppUrlsDifferent()) {
+            Assumption.fail("App URLs defined in test.properties and build.properties must be different, "
+                            + "and neither one can be a substring of the other, e.g localhost:8888 and "
+                            + "localhost:88889 are not accepted as two different app URLs.");
```

Comment:
you could say here "... e.g localhost:8888 as url in test.properties, and localhost:88889 as url in build.properties is not an acceptable combination."